src/pybind/mgr/balancer/module.py

```diff
@@ -627,9 +635,11 @@ def serve(self):
                     final = [int(p) for p in final]
                     final = [pool_name_by_id[p] for p in final if p in pool_name_by_id]
                 plan = self.plan_create(name, osdmap, final)
+                optimizing = True
```

Comment:
you need self.optimizing here and line 642 - python doesn't implicitly refer to member variables